lib/hathitrust/subjects.rb

```diff
@@ -0,0 +1,61 @@
+## frozen_string_literal: true
+
+module Traject::Macros::HathiTrust
+  module Subject
+
+    LOWER_LETTER_PAT = /\p{Lower}/
+    STRIP_PUNCT_PAT = /\A\p{Punct}*(.*?)\p{Punct}*\Z/
+
+    # Strip leading/trailing punctuation
+    # @param [String] string of a subject
+    # @return [String] The normalized subject string
+    def self.normalize_subject(str)
+      STRIP_PUNCT_PAT.match(str)[1].gsub("/\s+/", ' ')
+    end
+
+    # Determin the 880 (linking fields) for the given field. Should probably be pulled
+    # out into a more generically-available macro
+    # @param [MARC::Record] r The record
+    # @param [MARC::DataField] f The field you want to try to match
+    # @param [Array<MARC::DataField>] A (possibly empty) array of linked fields
+    def self.eight_eighties_for(r, f)
+      vid = f['6']
+      if vid
+        r.fields('880').select { |eef| eef['6']&.start_with? "#{f.tag}-#{vid.split("-").last}" }
+      else
+        []
+      end
+    end
+
+    # Get all the subfields of the given field and join them together with '--'
+    # @param [MARC::DataField] field
+    # return [String] a '--' delimited string of all  the lettered subfields
+    def self.subject_string(field)
+      self.normalize_subject(field.subfields.select { |sf| LOWER_LETTER_PAT.match(sf.code) }.map(&:value).join("--"))
```

Comment:
Could we call that "data subfields" rather than referring to "LOWER_LETTER_PAT"?